### DISCUSSION TITLE
How can I send the user to a timeout?

### DISCUSSION THREAD

**Discussion by IIDarkNessYT** [Rapptz/discord.py#8232] [Q&A] [answered]
I make my bot for moderation and not only. And I want to make a timeout instead of issuing a mute, how can I implement this on the developer version?

**Comment by trevorflahardy** [comment, marked as answer]
## Timing out a Member
Implementing a timeout on a `discord.Member` is pretty simple. Please note you can not timeout an instance of `discord.User` because users are not guild specific. You have two methods you can call to timeout the member.

1. [Member.timeout](https://discordpy.readthedocs.io/en/latest/api.html#discord.Member.timeout).
2. [Member.edit](https://discordpy.readthedocs.io/en/latest/api.html#discord.Member.edit) with the `timed_out_until` parameter.

### Small Example
```python
@bot.command()
@commands.has_permissions(moderate_members=True)
async def timeout(ctx: commands.Context, member: discord.Member):
    await member.timeout(datetime.timedelta(hours=1), reason=f"Requested by {ctx.author}")
    await ctx.send(f"I've timed out {member.mention} for 1 hour.")
```

### **How do I remove the timeout on the Member?**
Call either of the two methods above with the `until` or `timed_out_until` parameter as `None`

### **I need some more help?**
Feel free to join the [discord.py Discord server](https://discord.gg/dpy) for faster help in one of the help channels :)

**Reply by IIDarkNessYT** [reply]
Thank you!

**Comment by sgysh3nka** [comment]
Here you go!
```py
@bot.command()
async def mute(ctx, member : discord.Member, *, reason):
	s = ctx.author
	if not s.guild_permissions.administrator:
		return await ctx.channel.send(f'<@{s.id}> you don't have permissions!')
	if reason == None:
		await member.timeout(datetime.timedelta(hours=1), reason="No reason was given.")
	await member.timeout(datetime.timedelta(hours=1), reason=reason)
	await ctx.send(f'Violator {member.mention}, was timeouted.')
```
